src/transformers/data/test_generation_utils.py

```diff
@@ -0,0 +1,95 @@
+import random
+import unittest
+
+import timeout_decorator
+
+from transformers import is_torch_available
+from transformers.file_utils import cached_property
+from transformers.testing_utils import require_torch
+
+
+if is_torch_available():
+    import torch
+
+    from transformers import (
+        MarianConfig,
+        MarianMTModel,
+    )
+
+
+@require_torch
+class GenerationUtilsTest(unittest.TestCase):
+    @cached_property
+    def config_and_model(self):
+        config = MarianConfig.from_pretrained("sshleifer/tiny-marian-en-de")
+        return config, MarianMTModel(config)
+
+    @require_torch
+    def test_postprocess_next_token_scores(self):
+        config, model = self.config_and_model
+        # Initialize an input id tensor with batch size 8 and sequence length 12
+        input_ids = torch.arange(0, 96, 1).view((8, 12))
+
```

Comment:
```suggestion
eos = config.eos_token_id
bs = input_ids.shape[0]

scores_that_should_be_inf = [[(i, 299) for i in range(bs)], 
+[(i, 54) for i in range(bs)] +[(1,24)], 
[(i, eos) for i in range(bs)], 
]


```
why only (1,24)?

Reply:
btw that suggestion wont work perfectly

Reply:
You are right - removed the unnecessary `@require_torch` and simplified the cached properties into 2 independent properties. For the second test, only the (1, 24) is `-inf` because the banned_token is a sequence: [23, 24]. Since the `input_ids` contain the token id 23 only on batch 1 (by construction), the token is only banned for this batch.